api/v1alpha1/proxmoxmachine_types_test.go

```diff
@@ -56,18 +59,24 @@ var _ = Describe("ProxmoxMachine Test", func() {
 	})
 
 	Context("VirtualMachineCloneSpec", func() {
-		It("Should not allow empty source node", func() {
+		It("Should not allow specifying format if full clone is disabled", func() {
 			dm := defaultMachine()
-			dm.Spec.SourceNode = ""
+			dm.Spec.Full = ptr.To(false)
 
-			Expect(k8sClient.Create(context.Background(), dm)).Should(MatchError(ContainSubstring("should be at least 1 chars long")))
+			Expect(k8sClient.Create(context.Background(), dm)).Should(MatchError(ContainSubstring("Must set full=true when specifying format")))
 		})
 
-		It("Should not allow specifying format if full clone is disabled", func() {
+		It("Should not allow specifying TemplateSelector with SourceNode and/or TemplateID", func() {
 			dm := defaultMachine()
-			dm.Spec.Full = ptr.To(false)
+			dm.Spec.TemplateSelector = &TemplateSelector{MatchTags: []string{"test"}}
+			Expect(k8sClient.Create(context.Background(), dm)).Should(MatchError(ContainSubstring("mutually exclusive")))
+		})
 
-			Expect(k8sClient.Create(context.Background(), dm)).Should(MatchError(ContainSubstring("Must set full=true when specifying format")))
+		It("Should not allow specifying TemplateSelector with empty MatchTags", func() {
+			dm := defaultMachine()
+			dm.Spec.TemplateSelector = &TemplateSelector{MatchTags: []string{}}
+
+			Expect(k8sClient.Create(context.Background(), dm)).Should(MatchError(ContainSubstring("should have at least 1 items")))
```

Comment:
I think there should also be a test for both TemplateID and TemplateSelector being unset.